### ISSUE TITLE
Add IP address as an attribute on VMs

### ISSUE THREAD

**Issue by ddelnano** [terra-farm/terraform-provider-xenorchestra#112]
This was previously discussed in #49. At the time, I wasn't sure there were workable solutions, however, I had an idea on how to implement this.

If a VM has XCP-ng / Xenserver guest tools, Terraform should be able to access the VMs ip address (the same way the xenserver packer builder does [here](https://github.com/ddelnano/packer-builder-xenserver/blob/master/builder/xenserver/common/step_wait_for_ip.go#L52-L68)).

We will have to change when the Terraform provider considers a VM as created. Currently it will consider it a success if the VM is in the [Running state](https://github.com/terra-farm/terraform-provider-xenorchestra/blob/master/client/vm.go#L245-L250). In the case that we have guest tools installed we will need to wait longer for the kernel to boot and for the agent to register. In order to decide on which behavior to have, I think we should introduce a new attribute `wait_for_ip`. This attribute will be a bool and will only work if a VM has guest tools installed.

Below is an example of how the resource would look if the user wants to have the VM's IP address information accessible in Terraform code.

```
data "xenorchestra_network" "net" {
  name_label = "Pool-wide network associated with eth0"
}

resource "xenorchestra_vm" "vm_with_ip" {
  ...
  ...
  network {
      network_id = data.xenorchestra_network.net.id
  }
  wait_for_ip = true
}

# The value will be accessible like this
output "instance_ip_addr" {
  value = [
    xenorchestra.vm.vm_with.ip.network[*].ip_address
  ]
}
```

**Comment by ddelnano** [comment]
This could actually be accomplished in two parts. I didn't realize that the XO api returns these ip addresses in the "addresses" key on the `xo.getAllObject` response.

```
ddelnano@ddelnano-desktop:~/go/src/github.com/ddelnano/terraform-provider-xenorchestra$ xo-cli xo.getAllObjects filter='json:{"id":"d2efe162-35b3-f84f-8a59-6064b6875b61"}'
{
  "d2efe162-35b3-f84f-8a59-6064b6875b61": {
    "type": "VM",
    "addresses": {
      "0/ipv6/0": "fe80::d4d3:8aff:fe2d:9a4f",
      "0/ip": "192.168.88.86",
      "0/ipv4/0": "192.168.88.86"
    },

```

The tuple specified above is device number / proto / _something_. I still need to track down what that "something" is.

**Comment by ddelnano** [comment]
This format is documented [here](https://github.com/vatesfr/xen-orchestra/blob/54a85a8dd04fb22781a93b62aaaf66098115194b/packages/xo-server/src/_extractIpFromVmNetworks.js#L1-L3). This confirms that the first item in the tuple is the network interface number but I'm still unsure of what the final number is.

**Comment by ddelnano** [comment]
This will be released in v0.15.0 of the provider.
